### PR TITLE
refactor(types): fix typescript types

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

Github issue: #732 

This PR introduces several updates and fixes to enhance the project:

1. **Dependency Upgrades:**
   - Upgraded all devDependencies to their latest versions for improved performance, security, and compatibility.
  
2. **Node Version Update:**
   - Updated the `.node-version` file to utilize Node.js LTS version `18.19.0` for better stability and long-term support.

3. **Test File Updates:**
   - Updated test files to seamlessly function with newer versions of associated libraries.

4. **Workflow Fixes:**
   - Resolved issues within the Storybook-test workflow.

5. **TypeScript Enhancement:**
   - Addressed TypeScript type errors specifically related to the Tabs component.

Moreover, if encountering TypeScript errors with the updated version, consider adding the following configuration to your `tsconfig` file:

```json
"compilerOptions": {
  "paths": {
    "react": ["node_modules/@types/react"]
  }
}
```

You'll also need to install the latest React and ReactDOM types to prevent TypeScript errors.

-  npm: 
`npm i @types/react@18 @types/react-dom@18`

- yarn: 
`yarn add @types/react@18 @types/react-dom@18`

### CODE REVIEW

**Review by tihuan — APPROVED** [review]
LGTM!! Thank you for testing this with CZID and Cell Science repos directly 🔥 So exciting!!

File: .node-version [hunk 1]
```diff
@@ -1 +1 @@
-16.15.1
+18.19.0
```
Comment (masoudmanson) [inline .node-version:1]: Some packages, such as the latest version of Stylelint, require Node version >= 18. This version is an LTS version and works seamlessly with the rest of the project.

File: lerna.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "$schema": "node_modules/lerna/schemas/lerna-schema.json",
   "useNx": true,
-  "useWorkspaces": true,
```
Comment (masoudmanson) [inline lerna.json:4]: Workspaces are used by default and useWorkspaces has been removed.

File: packages/components/src/core/styles/common/constants/spaces.ts [hunk 3]
```diff
@@ -8,4 +7,5 @@ export enum Spaces {
   L = "14px",
   XL = "22px",
   XXL = "38px",
+  DEFAULT = M,
```
Comment (masoudmanson) [inline packages/components/src/core/styles/common/constants/spaces.ts:10]: Typescript complained about having duplicate values.

**Pull request merged**